File: Daily Test/C/Rising Sequence - Minimum Addition 30.7.2025.c
#include <stdio.h>
#include <math.h>

int main() {
    int n, k, i, c = 0;
    scanf("%d", &n);
    int a[n];
    for(i=0;i<n;i++) scanf("%d", &a[i]);
    scanf("%d", &k);
    int p = a[0];
    for(i=1;i<n;i++) {
        if(a[i] <= p) {
            int x = ceil((p - a[i] + 1.0)/k);
            a[i] += x * k;
            c += x;
        }
        p = a[i];
    }
    printf("%d\n", c);
    return 0;
}

File: Daily Test/Java/Rising Sequence - Minimum Addition 30.7.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for(int i=0;i<n;i++) a[i] = sc.nextInt();
        int k = sc.nextInt(), c = 0, p = a[0];
        for(int i=1;i<n;i++) {
            if(a[i] <= p) {
                int x = (int)Math.ceil((p - a[i] + 1.0) / k);
                a[i] += x * k;
                c += x;
            }
            p = a[i];
        }
        System.out.println(c);
    }
}